src/ResourceManager/IotCentral/Commands.IotCentral/IotCentral/NewAzureRmIotCentralApp.cs

```diff
@@ -106,6 +103,28 @@ public override void ExecuteCmdlet()
 
         }
 
+        private void CheckNameAvailability()
+        {
+            var checkNameInputs = new OperationInputs(this.Name, "IoTApps");
+            var nameAvailabilityInfo = this.IotCentralClient.Apps.CheckNameAvailability(checkNameInputs);
+            this.CheckAvailability(nameAvailabilityInfo);
+        }
+
+        private void CheckSubdomainAvailability()
+        {
+            var checkSubdomainInputs = new OperationInputs(this.Subdomain, "IoTApps");
+            var subdomainAvailabilityInfo = this.IotCentralClient.Apps.CheckSubdomainAvailability(checkSubdomainInputs);
+            this.CheckAvailability(subdomainAvailabilityInfo);
+        }
+
+        private void CheckAvailability(AppAvailabilityInfo availabilityInfo)
```

Comment:
I would rename these from *Check* to something more clear, *EnsureAvailableOrThrow* for all three of these methods would make it easier to tell what will happen